elmslie-rxjava-3/src/main/java/vivid/money/elmslie/core/MappingActorCompat.kt

```diff
@@ -0,0 +1,221 @@
+package vivid.money.elmslie.core
+
+import io.reactivex.rxjava3.core.Completable
+import io.reactivex.rxjava3.core.Maybe
+import io.reactivex.rxjava3.core.Observable
+import io.reactivex.rxjava3.core.Single
+import vivid.money.elmslie.core.store.MappingActor
+
+/**
+ * Contains internal event mapping helpers for RxJava3
+ */
+@Suppress("ComplexInterface", "TooManyFunctions")
+interface MappingActorCompat<Event : Any> : MappingActor<Event> {
+
+    fun Completable.mapEvents(
+        completionEvent: Event? = null,
+        errorMapper: (Throwable) -> Event? = { null }
+    ): Observable<Event> = toObservable<Event>().mapEvents({ null }, errorMapper, completionEvent)
+
+    fun <T : Any> Single<T>.mapEvents(
+        eventMapper: (T) -> Event? = { null },
+        errorMapper: (Throwable) -> Event? = { null },
+        completionEvent: Event? = null,
+    ): Observable<Event> = toObservable().mapEvents(eventMapper, errorMapper, completionEvent)
+
+    fun <T : Any> Maybe<T>.mapEvents(
+        eventMapper: (T) -> Event? = { null },
+        errorMapper: (throwable: Throwable) -> Event? = { null },
+        completionEvent: Event? = null,
+    ): Observable<Event> = toObservable().mapEvents(eventMapper, errorMapper, completionEvent)
+
+    fun <T : Any> Observable<T>.mapEvents(
+        eventMapper: (T) -> Event? = { null },
+        errorMapper: (throwable: Throwable) -> Event? = { null },
+        completionEvent: Event? = null,
+    ): Observable<Event> = flatMapMaybe { Maybe.fromCallable<Event> { eventMapper(it) } }
+        .switchIfEmpty(Maybe.fromCallable<Event> { completionEvent }.toObservable())
+        .doOnNext { it.logSuccessEvent() }
+        .onErrorResumeNext { Observable.fromIterable(listOfNotNull(it.logErrorEvent(errorMapper))) }
+
+
+    @Deprecated(
+        "Please, use the default mapEvents method",
+        ReplaceWith("mapEvents({ successEvent }, { errorEvent })")
+    )
+    fun Completable.mapEvents(
+        successEvent: Event,
+        errorEvent: Event
+    ): Observable<Event> = mapEvents(successEvent) { errorEvent }
+
+    @Deprecated(
+        "Please, use the default mapEvents method",
+        ReplaceWith("mapEvents()")
+    )
+    fun Completable.ignoreEvents(): Observable<Event> = mapEvents()
+
+    @Deprecated(
+        "Please, use the default mapEvents method",
+        ReplaceWith("mapEvents({ successEvent })")
+    )
+    fun Completable.mapSuccessEvent(
+        successEvent: Event
+    ): Observable<Event> = mapEvents(successEvent)
+
+    @Deprecated(
+        "Please, use the default mapEvents method",
+        ReplaceWith("mapEvents(errorMapper = { errorEvent })")
+    )
+    fun Completable.mapErrorEvent(
+        errorEvent: Event
+    ): Observable<Event> = mapEvents(null, { errorEvent })
```

Comment:
here are the breaking changes, the old implementation use `::onErrorReturn` and passes the success event as is, but the current implementation omit the success event and emit only errors